### PR TITLE
feat/table

### PR DESCRIPTION
**Pull request by kacperborowieckb**

### CODE REVIEW

File: src/components/StudentsTable.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  getDefaultStudentsColDef,
+  getStudentsColDefs,
+  type StudentsTableRowData
+} from '@/helpers/columnDefinitions'
+import { createGridError } from '@/utils/createGridError'
+
+const studentsStore = useStudentsStore()
+
+const studentsColumnDefs = ref(getStudentsColDefs())
+const defaultStudentsColumnDef = ref(getDefaultStudentsColDef())
```
Comment (l-gorka) [inline src/components/StudentsTable.vue]: do we need reactivity on this property? also why you decided to use function to create this object? I think default column def should be constant

File: src/components/StudentsTable.vue [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const studentsTableApi = shallowRef<GridApi<StudentsTableRowData> | null>(null)
```
Comment (l-gorka) [inline src/components/StudentsTable.vue:28]: why shallow ref here?

File: src/helpers/columnDefinitions.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  {
+    headerName: 'Final Grade',
+    field: 'finalGrade'
+  },
+  {
+    headerName: 'Hobbies',
+    field: 'hobbies',
+    valueGetter: studentsHobbiesGetter
+  }
+]
```
Comment (l-gorka) [inline src/helpers/columnDefinitions.ts]: not sure why we need a functions here, there's nothing to compute

File: src/utils/createGridError.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import type { GridApi } from 'ag-grid-community'
+
+export const createGridError = (gridApi: GridApi, errorMessage: string) => {
+  gridApi.setGridOption('overlayNoRowsTemplate', errorMessage)
+  gridApi.showNoRowsOverlay()
+}
```
Comment (l-gorka) [inline src/utils/createGridError.ts]: in real app we need to show/hide this error conditionally so probably we need another func to hide it, much overhead

I think composable taking gridApi as an argument, and exposing hide/show would fit here better

File: src/components/StudentsTable.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<template>
+  <AgGridVue
+    :columnDefs="studentsColumnDefs"
+    :defaultColDef="defaultStudentsColumnDef"
+    :rowData="studentsStore.students"
+    @gridReady="onGridReady"
+    class="students-table ag-theme-quartz-dark"
```
Comment (l-gorka) [inline src/components/StudentsTable.vue]: static binding should go first

File: src/components/StudentsTable.vue (hunk 2 above)
Comment (kacperborowieckb) [reply src/components/StudentsTable.vue:28]: I found something like this: https://www.ag-grid.com/vue-data-grid/grid-interface/#api-within-events-and-callbacks

File: src/helpers/valueGetters.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import type { ValueGetterParams } from 'ag-grid-community'
+
+import type { StudentsTableRowData } from '@/helpers/columnDefinitions'
+
+type StudentsGetterParams<T> = ValueGetterParams<StudentsTableRowData, T>
+
+export const studentsHobbiesGetter = (params: StudentsGetterParams<string[]>) => {
+  return params.data?.hobbies?.length ? params.data.hobbies.join(', ') : 'No hobbies.'
```
Comment (DrkNowak) [inline src/helpers/valueGetters.ts]:
```
const hobbies =  params.data?.hobbies || []

return hobbies.join(', ') || 'No hobbies';
```

or eventually just extract hobbies to a variable to keep cleanier code

Comment (DrkNowak) [reply src/helpers/valueGetters.ts]: + pls change function name to "getStudentHobbies". Pls apply everywhere. function name in our project always starts with a verb.

**Review by DrkNowak — APPROVED** [review]

**Review by l-gorka — APPROVED** [review]

**Pull request merged**